applications/nrf_desktop/src/modules/hid_forward.c

```diff
@@ -1258,34 +1215,126 @@ static void forward_hid_out_report(const struct hid_report_event *event,
 	}
 }
 
-static bool handle_hid_report_event(const struct hid_report_event *event)
+static struct subscriber *find_subscriber(const void *sub_id)
 {
-	/* Ignore HID input events. */
-	if (event->subscriber) {
-		return false;
+	struct subscriber *sub = NULL;
+
+	for (size_t i = 0; i < ARRAY_SIZE(subscribers); i++) {
+		if (subscribers[i].id == sub_id) {
+			sub = &subscribers[i];
+			break;
+		}
 	}
 
-	size_t sub_idx;
+	return sub;
+}
 
-	for (sub_idx = 0; sub_idx < ARRAY_SIZE(subscribers); sub_idx++) {
-		if (subscribers[sub_idx].id == event->source) {
-			break;
+static void send_empty_hid_out_reports(struct hids_peripheral *per, struct subscriber *sub)
```

Comment:
Overall looks good. Please don't move code: "send_empty_hid_out_reports", "handle_hid_report_event" etc. so that commit looks more clearly.

Reply:
I actually wanted to reorganize code a bit before integrating HID report queue utility (to improve readability). Scope of this PR was kept small to ensure we will be able to properly review it. I would prefer to keep functions in new places if possible.